pkg/crd/known_types.go

```diff
@@ -113,6 +111,53 @@ var KnownPackages = map[string]PackageOverride{
 	},
 }
 
+// ObjectMetaPackages overrides the ObjectMeta in all types
+var ObjectMetaPackages = map[string]PackageOverride{
+	"k8s.io/apimachinery/pkg/apis/meta/v1": func(p *Parser, pkg *loader.Package) {
+		// execute the KnowPackages for `k8s.io/apimachinery/pkg/apis/meta/v1` if any
+		if f, ok := KnownPackages["k8s.io/apimachinery/pkg/apis/meta/v1"]; ok {
+			f(p, pkg)
+		}
+		// This is an allow-listed set of properties of ObjectMeta, other runtime properties are not part of this list
+		// See more discussion: https://github.com/kubernetes-sigs/controller-tools/pull/395#issuecomment-691919433
+		p.Schemata[TypeIdent{Name: "ObjectMeta", Package: pkg}] = apiext.JSONSchemaProps{
+			Type: "object",
+			Properties: map[string]apiext.JSONSchemaProps{
+				"name": apiext.JSONSchemaProps{
+					Type: "string",
+				},
+				"namespace": apiext.JSONSchemaProps{
+					Type: "string",
+				},
+				"annotations": apiext.JSONSchemaProps{
+					Type: "object",
+					AdditionalProperties: &apiext.JSONSchemaPropsOrBool{
+						Schema: &apiext.JSONSchemaProps{
+							Type: "string",
+						},
+					},
+				},
+				"labels": apiext.JSONSchemaProps{
+					Type: "object",
+					AdditionalProperties: &apiext.JSONSchemaPropsOrBool{
+						Schema: &apiext.JSONSchemaProps{
+							Type: "string",
+						},
+					},
+				},
+				"finalizers": apiext.JSONSchemaProps{
```

Comment:
this is odd. Why finalizers?

Reply:
I had the same thought, I'd have expected either all the ObjectMeta fields or the non-system fields here https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/object-meta/

Reply:
@jessefeinman Can you open an issue to discuss this?

Reply:
https://github.com/kubernetes-sigs/controller-tools/issues/805